# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/WasmBCE.cpp
# Commit: ce799cb566ac
# Full Hash: ce799cb566ac0fb95468118608132be11db6a359
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-03-24 21:54:20
# Regressor Bug: 1642412
# File Overlap Count: 19
# Description:
#   Bug 1642412 - wasm: Rollup of inline table.get/set for anyref. r=lth
#   
#   Rollup of the previous inline table.get/set patches.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D141024
# ==============================================================================

diff -r c10609d434ae -r ce799cb566ac js/src/jit/WasmBCE.cpp
--- a/js/src/jit/WasmBCE.cpp	Thu Mar 24 17:40:32 2022 +0000
+++ b/js/src/jit/WasmBCE.cpp	Thu Mar 24 17:56:50 2022 +0000
@@ -45,7 +45,13 @@
           MWasmBoundsCheck* bc = def->toWasmBoundsCheck();
           MDefinition* addr = bc->index();
 
-          // Eliminate constant-address bounds checks to addresses below
+          // We only support bounds check elimination on wasm memory, not
+          // tables. See bug 1625891.
+          if (!bc->isMemory()) {
+            continue;
+          }
+
+          // Eliminate constant-address memory bounds checks to addresses below
           // the heap minimum.
           //
           // The payload of the MConstant will be Double if the constant